impl/client/src/test/javascript/pentaho/data/filter/isLike.spec.js

```diff
@@ -0,0 +1,142 @@
+
+        var filter  = new IsLikeFilter({property: "name", value: "tes", anchorStart: true, anchorEnd: false});
+        var negFilter = new IsLikeFilter({property: "name", value: "es", anchorStart: true, anchorEnd: false});
+
+        var result = filter.contains(elem);
+        var negResult = negFilter.contains(elem);
+
+        expect(result).toBe(true);
+        expect(negResult).toBe(false);
+      });
+
+      it("should return `true` if `elem` has property `property` with a value matching a suffix of `value`", function() {
+
+        var filter  = new IsLikeFilter({property: "name", value: "est", anchorStart:false, anchorEnd: true});
+        var negFilter = new IsLikeFilter({property: "name", value: "es", anchorStart: false, anchorEnd: true});
+
+        var result = filter.contains(elem);
+        var negResult = negFilter.contains(elem);
+
+        expect(result).toBe(true);
+        expect(negResult).toBe(false);
+      });
+
+      it("should return `true` if `elem` has property `property` with a value exactly matching `value`", function() {
+
+        var filter  = new IsLikeFilter({property: "name", value: "test", anchorStart:true, anchorEnd: true});
+        var negFilter = new IsLikeFilter({property: "name", value: "tes", anchorStart: true, anchorEnd: true});
+        var negFilter2 = new IsLikeFilter({property: "name", value: "est", anchorStart: true, anchorEnd: true});
+
+        var result = filter.contains(elem);
+        var negResult = negFilter.contains(elem);
+        var negResult2 = negFilter2.contains(elem);
+
+        expect(result).toBe(true);
+        expect(negResult).toBe(false);
+        expect(negResult2).toBe(false);
+      });
+    }); // #contains
+
+    describe("#contentKey", function() {
+
+      it("should return '(like propName valueKey anchorS anchorE)'", function() {
+        var filter  = new IsLikeFilter({property: "name", value: "str"});
+        var anotherFilter = new IsLikeFilter({property: "name", value: "str", anchorStart: true, anchorEnd: true});
+
+        expect(filter.contentKey).toBe("(like name str false false)");
+        expect(anotherFilter.contentKey).toBe("(like name str true true)");
+      });
+
+      it("should return '(like propName  false false)' when no value is set'", function() {
+        var filter  = new IsLikeFilter({property: "name"});
+
+        expect(filter.contentKey).toBe("(like name  false false)");
+      });
+
+      it("should return '(like  valueKey false false)' when no property is set'", function() {
+        var filter  = new IsLikeFilter({value: "str"});
+
+        expect(filter.contentKey).toBe("(like  str false false)");
+      });
+
+      it("should return '(like   false false)' when no property or value are set'", function() {
+        var filter  = new IsLikeFilter();
+
+        expect(filter.contentKey).toBe("(like   false false)");
```

Comment:
yes, missing values are represented as empty spaces, otherwise the content key of a filter with two (or more) optional parameters could not distinguish between having the first parameter empty and the second defined and vice-versa.